### ISSUE TITLE
Nextcloud entry in Windows Explorer context menu doesn't show up for virtual files

### ISSUE THREAD

**Issue by getle87** [nextcloud/desktop#3584]
### How to use GitHub

-    Please use the 👍 reaction to show that you are affected by the same issue.
-    Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
-    Subscribe to receive notifications on status change and new comments.

### Actual behaviour

The Nextcloud entry in the context menu doesn't appear when selecting a virtual file (VFS Share)
![fig1](https://user-images.githubusercontent.com/58705292/126618924-0259355d-1a4e-4238-8f3d-8f41e8819a4a.png)

Downloaded files (Always keep on this device) oder a virtual folders aren't affected from this issue.
![fig2](https://user-images.githubusercontent.com/58705292/126619539-e8ea6518-f6ae-4da7-a765-1f32c19dc7c8.png)

![fig3](https://user-images.githubusercontent.com/58705292/126619421-457d7b2b-0ec8-4b03-b39d-f26f0afdfe7e.png)

### Expected behaviour

The Nextcloud entry in the context menu should be always available.

### Client configuration

Client version: 3.2.4

Operating system: Windows 10 Pro 20H2

OS language: DE

### Server configuration

Nextcloud version: 20.0.9

**Comment by claell** [comment]
I can confirm this behavior, although it looked more random to me, whether a context menu is displayed or not. But maybe you are right and it is caused by virtual vs downloaded files.

**Comment by claell** [comment]
(I think, I experienced this problem with folders mainly, not tested with files)

**Comment by kenvix** [comment]
Same problem. The nextcloud context menu is not available for online-only (not hydrated) files.

**Comment by claell** [comment]
Push

**Comment by mgallien** [comment]
@getle87 can you test again with latest release ?
in case this is still broken, can you share your desktop client log ?

**Comment by getle87** [comment]
I have set up a test environment (3 nc servers for federation tests, 2 clients) for further tests with this an other issues. The problem still persists in version 3.2.2. I attached 4 screenshot and the debug archive (zip file).

virtual file:
![vfs_file](https://user-images.githubusercontent.com/58705292/131847517-7bfe1de5-721b-4bdb-9cef-38fa8ee60acf.png)
local file
![local_file](https://user-images.githubusercontent.com/58705292/131847537-a63ee661-88be-42d5-89fa-c5de90f8a82c.png)
virtual folder
![vfs_folder](https://user-images.githubusercontent.com/58705292/131847520-9b06dd0e-b050-47e1-a6be-c111ab71f3a0.png)
local folder
![local_folder](https://user-images.githubusercontent.com/58705292/131847540-a6183d6e-b279-4c1c-b40d-101a3e964edc.png)

Log Archive
[debug.zip](https://github.com/nextcloud/desktop/files/7099055/debug.zip)

**Comment by getle87** [comment]
The problem still persists in version 3.3.3

**Comment by Trafo** [comment]
I created a fix #3799 . This bug should be only reproducible with newer .msi installers.

**Comment by getle87** [comment]
Problem still exists in 3.3.4

**Comment by getle87** [comment]
I can confirm, the issue is fixed in the current version (3.4.1)

**Comment by marcteys** [comment]
I still have the issue on 3.4.1 for Windows 10 . It is happening mainly on very large folders.

**Comment by fraktal-sb** [comment]
issue still not solved; environment Windows 11; nexcloud version 3.5.1; see Screencopy below

![2022-06-13 08_51_10-Clipboard](https://user-images.githubusercontent.com/69430537/173296913-1ad45e62-826e-49c4-bd6b-13b6138d70e4.png)

**Comment by NudiDesign** [comment]
![image](https://user-images.githubusercontent.com/99628796/211243377-ac3fcbd8-bae1-4ff0-b424-0ce8bfceb90b.png)

Still causing problems on v3.6.4

**Comment by davidpasquale** [comment]
Still get the same random error.
For some folders, even if the subfiles and subfolders inside that folder show the nextcloud menu, the main folder does not, therefore the full nextcloud virtual folder does NOT show the menu as well!!!

Nextcloud 27.0
Desktop Nextcloud 3.9 (windows)

Please fix or give the reason why this is happening!

Thank you

**Comment by gaticssa** [comment]
Seeing this error as well.

Example:
```
GroupFolder
	folderA
	folderB
	folderC
```
The context menu entry is missing for folderB, but is visible for folderA and folderC. The current user has the same (read only) permissions on all three folders.

So far, the broken folders appear to be the same for every user, meaning the context menu entry for a specific folder either shows up for everyone, or for nobody at all.

I can't see any pattern why some folders are ok and others are broken.

This is a fresh install (both client and server) and none of the listed folders are hydrated or have ever been hydrated on the client.

Client: Windows 10 22H2, Nextcloud Desktop 3.9.0/3.9.1
Server: Nextcloud Hub 5 (27.0.1), Ubuntu, PHP 8.1.21, mysql 10.3.39

**Comment by NudiDesign** [comment]
Seems to be a pattern forming here, around empty folders. Using gaticssa's example, if folderB is empty then I get the context menu on folderA and folderC, but not on folderB or GroupFolder (or anything further up the tree).

**Comment by davidpasquale** [comment]
@NudiDesign , I do not agree, fo my case it is happening for only folders with files inside.

**Comment by gaticssa** [comment]
> Seems to be a pattern forming here, around empty folders. Using gaticssa's example, if folderB is empty then I get the context menu on folderA and folderC, but not on folderB or GroupFolder (or anything further up the tree).

Just to be clear: The folders in my example are not empty either. In fact, they all have lots of subfolders.

**Comment by NudiDesign** [comment]
What I was seeing was if any subfolder in a tree was empty, then it and 
every other folder up the tree from it would have the menu missing.

But now (thanks folks) I've gone back and checked and I guess it was a 
red herring - I'm getting the menu on empty folders now.

------ Original Message ------
From: "gaticssa" ***@***.***>
To: "nextcloud/desktop" ***@***.***>
Cc: "NudiDesign" ***@***.***>; "Mention" 
***@***.***>
Sent: 30/07/2023 7:32:41 PM
Subject: Re: [nextcloud/desktop] Nextcloud entry in Windows Explorer 
context menu doesn't show up for virtual files (#3584)

**Comment by elschman** [comment]
I have nextcloud 3.9 and 3.10 desktop client installed. 
Same issue still existing.
There are two folders both are shared both are not empty and containing a lot of files and directories. all are virtual and not downloaded.
I do see one folder showing the nextcloud context menu and one folder not.
There is also no desktop.ini in the folders. Sync was done right before this writing. I guess it should be reopened.

**Comment by joshtrichards** [comment]
There is an existing issue tracking this for >v3.6.0: #5008. Those of you still experiencing this behavior, please see if that report seems to apply (or open a new issue if not).

The OP of this issue has already confirmed their matter fixed. Whatever is still going on for the rest of you seems to have a different root cause than in this (closed) Issue.

**Comment by hermeneuticlens** [comment]
I am using Nextcloud client latest 3.14.1 on Windows 11. I followed all your steps and still don't work. Only folders and already locally available files have Nextcloud in the context menu. Online only files do not have Nextcloud in the context menu. The issue has NOT been resolved.
